self-core-impl/src/main/java/com/selfxdsd/core/GithubStars.java

```diff
@@ -0,0 +1,88 @@
+package com.selfxdsd.core;
+
+import com.selfxdsd.api.Stars;
+import com.selfxdsd.api.storage.Storage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.json.Json;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Github stars.
+ */
+final class GithubStars implements Stars {
+
+    /**
+     * Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(
+            GithubStars.class
+    );
+
+    /**
+     * Github repo stars base uri.
+     */
+    private final URI starsUri;
+
+    /**
+     * Github's JSON Resources.
+     */
+    private final JsonResources resources;
+
+    /**
+     * Self storage, in case we want to store something.
+     */
+    private final Storage storage;
+
+    /**
+     * Ctor.
+     *
+     * @param resources Github's JSON Resources.
+     * @param starsUri Stars base URI.
+     * @param storage Storage.
+     */
+    GithubStars(
+            final JsonResources resources,
+            final URI starsUri,
+            final Storage storage
+    ) {
+        this.resources = resources;
+        this.starsUri = starsUri;
+        this.storage = storage;
+    }
+
+    @Override
+    public boolean star() {
```

Comment:
@ak2703 Please add a puzzle for the unit testing of this method. Normally, I would reject a PR without unit tests, but this is a simple usecase.

A "puzzle" is a ``todo`` and in this case it should like this:

```java
@todo #630:30min Please provide some unit tests for method star() here.
 We should start a mock server and catch the incoming request to assert
 that it is constructed correctly.
```

You can add it in the javadoc of the class. Please pay attention to its grammar: lines 2..n should start with a space (they should be aligned with the ``t`` instead of the ``@``).